src/PuliPluginImpl.php

```diff
@@ -141,9 +150,39 @@ public function postAutoloadDump()
 
         $factoryFile = Path::makeAbsolute($factoryFile, $this->rootDir);
 
-        $this->insertFactoryClassConstant($autoloadFile, $factoryClass);
-        $this->insertFactoryClassMap($classMapFile, $vendorDir, $factoryClass, $factoryFile);
-        $this->setBootstrapFile($autoloadFile);
+        $autoload = $this->composer->getPackage()->getAutoload();
+        $autoload['classmap'][] = $factoryFile;
+        $this->composer->getPackage()->setAutoload($autoload);
+
+        if (!file_exists($factoryFile)) {
+            $filesystem = new Filesystem();
+            $filesystem->dumpFile($factoryFile, '<?php class '.$factoryClass.' {}');
```

Comment:
Is this here to avoid a Class not found problem? If it is, I think there should be a comment explaining it as it seems we are creating an empty class, which is weird for a generated factory.